src/OpenRiaServices.Hosting.AspNetCore/Framework/AspNetCore/Serialization/ArrayPoolStream.cs

```diff
@@ -116,7 +119,21 @@ private void EnsureBufferCapacity()
             // Ensure we never return buffer twice in case TakeBuffer below throws
             _buffer = null;
 
-            int nextSize = Math.Min(_position * 2, _maxSize);
+            int nextSize = _position * 2;
+            if (nextSize < 0)
```

Comment:
It might be good with a comment about why/when we get negative numbers. Such as when size is above 1GB we can get a numeric overflow and that we limit the total size to int.Max bytes